### PR TITLE
HTTPProxy API Spec update

### PR DESCRIPTION
**Pull request by stevesloka**
As HTTPProxy progresses, this PR again changes the design of the HTTPProxy CRD spec. 

**Note:** Mostly all the code/tests are commented out to keep the build passing. =)

### CODE REVIEW

**Comment by stevesloka** [comment]
My only struggle with the changes is related to `HeaderOnly`. I don't like the name, but `Header-->Header` also felt weird in the yaml.

File: apis/projectcontour/v1alpha1/httpproxy.go [hunk 1]
```diff
@@ -36,18 +36,30 @@ type Include struct {
...
 // Condition are policies that are applied on top of HTTPProxies.
 type Condition struct {
 	// Prefix defines a prefix match for a request.
 	Prefix string `json:"prefix,omitempty"`
-	// HeadersMatch represent a set of HTTP headers that match the key/value exactly as specified.
-	HeadersMatch map[string][]string `json:"headersMatch,omitempty"`
-	// HeadersContain represent a set of HTTP headers that match the key exactly and the value as a contains.
-	HeadersContain map[string][]string `json:"headersContain,omitempty"`
+	// Header matches only the key of the header ignoring any value supplied in the request.
+	HeaderOnly string `json:"headerOnly,omitempty"`
```
Comment (youngnick) [inline apis/projectcontour/v1alpha1/httpproxy.go]: Maybe HeaderPresent?

Comment (stevesloka) [reply apis/projectcontour/v1alpha1/httpproxy.go]: I like that better and it's what Envoy calls it. 👍

**Comment by davecheney** [comment]
Merge in 01c0d94

**Pull request closed without merging**